### PR TITLE
Update README for release 0.7.1

### PR DESCRIPTION
**Pull request by mamoodi**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
No it does not fix any issues.

Updated the README getting started.
Update pyproject to 0.7.1

**Give a brief summary of what the PR does, explaining any non-trivial design decisions**

**Other references**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,31 +45,7 @@ OpenDevin agents collaborate with human developers to write code, fix bugs, and
...
-docker run -it \
-    --pull=always \
-    -e SANDBOX_USER_ID=$(id -u) \
-    -e WORKSPACE_MOUNT_PATH=$WORKSPACE_BASE \
-    -v $WORKSPACE_BASE:/opt/workspace_base \
-    -v /var/run/docker.sock:/var/run/docker.sock \
-    -p 3000:3000 \
-    --add-host host.docker.internal:host-gateway \
-    --name opendevin-app-$(date +%Y%m%d%H%M%S) \
-    ghcr.io/opendevin/opendevin:0.7
```
Comment (xingyaoww) [inline README.md:67]: How about we still keep this command but change the version to `latest` here? Personally. I still feel having a command to copy and paste on the `README.md` is kinda convenient.

Comment (mamoodi) [reply README.md:67]: You want the entire section to remain in the README? Do you want me to update it to be exactly the same as the documentation?

Comment (tobitege) [reply README.md:67]: We just had the docs updated earlier. I'd prefer the main docker command to stay as well.

Comment (mamoodi) [reply README.md:67]: The docs right now are not exactly the same as the README.
This PR: https://github.com/OpenDevin/OpenDevin/pull/2804 only changed it in the docs but not in the README. 

Should we keep everything the same as right now in the README and just remove the 0.7 tag? Or should we align it to be exactly as the docs?

Comment (tobitege) [reply README.md:67]: I wouldn't mind the README to be the same as the content in 2804. the extra explanations should avoid some confusion for new users.

Comment (mamoodi) [reply README.md:67]: Agreed. 
The problem I'm trying to solve is the duplication. Just like the PR, one of the places keeps getting missed. I'll just keep it as is right now until we can come. Just removed the tag

Comment (tobitege) [reply README.md:67]: Agree with you, too. The de-duplication might just be a feature we aren't using yet with the docs?
Good task for a "good first issue" PR maybe.

**Review by tobitege — APPROVED** [review]
LGTM

**Comment by tobitege** [comment]
Oh you might have to run "poetry lock --no-update" and include that too @mamoodi

**Pull request merged**